### ISSUE TITLE
Add React context to enable/disable animations from this lib

### ISSUE THREAD

**Issue by tristanjohnson849** [tristanjohnson849/react-controlled-animations#23]
Lib users may want to disable animations for accessibility or for user control. Should have a context/provider that allows lib users to control whether animations are run in this lib.

Context could be referenced from useAnimatedRef, where if disabled=true, we run a useEffect to call onAnimationEnd on the next render. If possible, force currentAnimation back to null, but could be tricky

**Comment by tristanjohnson849** [comment]
After more thinking, the useAnimatedRef shouldn't try to manipulate currentAnimation. The onAnimationEnd will do this correctly for the async hooks that pass endTransition, but one-way-flow means useAnimatedRef should just ignore currentAnimation but accept whatever's given in other cases
